torch_geometric/nn/models/neural_fingerprint.py

```diff
@@ -0,0 +1,73 @@
+import torch
+import torch.nn as nn
+
+from torch_geometric.nn import MessagePassing
+from torch_geometric.utils import add_self_loops
+
+
+class NeuralConv(MessagePassing):
+    def __init__(self, num_features, fingerprint_length):
+        super(NeuralConv, self).__init__(aggr="add")
+        self.H = nn.ModuleList()
+        # H[i] would correspond to the hidden weight which would be used
+        # when num_neighbors=i+1
+        for i in range(5):
+            self.H.append(nn.Linear(num_features, num_features))
+
+        self.W = nn.Linear(num_features, fingerprint_length)
+
+    def forward(self, x, edge_index):
+        edge_index = add_self_loops(edge_index)
+
+        return self.propagate(edge_index, size=(x.size(0), x.size(0)), x=x)
+
+    def message(self, x_i, x_j, edge_index, size):
+        num_neighbors = x_j.size(0)
+
+        return {"msg": x_j, "num_neighbors": num_neighbors}
+
+    def update(self, aggr_out):
+        num_neigbors = aggr_out["num_neighbors"] - 2
+        msg = aggr_out["msg"]
+        updated_features = self.H[num_neigbors](msg)
+        updated_features = nn.Sigmoid()(updated_features)
+        updated_fingerprint = self.W(updated_features).softmax(dim=-1)
+        return updated_features, updated_fingerprint
+
+
+class NeuralFingerprint(nn.Module):
+    """The NeuralFingerprint is the model suggested in the paper
+    <https://arxiv.org/pdf/1509.09292.pdf>
+    The model generates fingerprints for molecules.
+
+    Args:
+        num_features(int): The number of features of the atom in the data.
+        fingerprint_length(int): Length of the fingerprint generated.
+        radius(int): Radius is the number of layers in the model.
```

Comment:
```suggestion
        in_channels(int): The number of features of the atom in the data.
        out_channels(int): Length of the fingerprint generated.
        num_layers(int): Radius is the number of layers in the model.
```
Make docs consistent with other classes in `models` like https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.models.GCN.html#torch_geometric.nn.models.GCN

Reply:
@wsad1 Thanks for your review, I will change the max_degree thing and set it to default 5. Sorry for the trivial question but can you please guide me as to what should be in a test file, I am new to contributing.

Reply:
For this model you can test two things:
1. Test the model with random inputs and 2 different `num_layers`. 
2. Add a `__repr__` function and test the output of that too.
You can refer to these tests: https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/models/test_rect.py and https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/models/test_rev_gnn.py

Reply:
@wsad1 Thanks for the guidance, I will start working on the tests.

Reply:
I apologise for the delay, currently I am into my final exams of the semester, I will try to add the tests as soon as possible.

Reply:
Sorry for the delay, I am now done with my exams, I Will try to add the tests by this weekend.

Reply:
Yes, this is what we usually do. You can look up other test files of models as guidance. It doesn't need to be complex.

Reply:
@rusty1s @wsad1 I am starting to realise that there may be many problems with my implementation. First of all I think I cannot return a dictionary from the message method.
The paper requires to apply a different weight depending on the number of neighbours of each node, due to my lack of understanding I thought that the message function deals with a single node at a time so I assumed that number of neighbours would be x_j.size(0), now I get it it's wrong.
Since I am beginner, I would request to please guide me as to how I can apply different weights to different nodes based on their degree.
I will do my best to implement this correctly, I just need some help.

Reply:
Oh wait, isn't this what we already do in https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/mf_conv.py?

Reply:
Thank you, I will look into this!